firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -107,12 +107,21 @@
   }
 
   /**
-   * Perform pre-condition checks to make sure {@link FirebaseOptions#getApiKey()} and {@link
-   * FirebaseOptions#getProjectId()} are not null.
+   * Perform pre-condition checks to make sure {@link FirebaseOptions#getApiKey()} and ({@link
+   * FirebaseOptions#getProjectId()} or {@link FirebaseOptions#getGcmSenderId()} ()} are not null.
    */
   private void preConditionChecks() {
+    Preconditions.checkNotNull(getApplicationId());
+    Preconditions.checkNotNull(getProjectIdentifier());
     Preconditions.checkNotNull(firebaseApp.getOptions().getApiKey());
```

Comment:
Wanna create a method for firebaseApp.getOptions().getApiKey()?
maybe #getApiKey?